src/cdk/tree/nested-node.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {
+  AfterContentInit,
+  ContentChildren,
+  Directive,
+  forwardRef,
+  Inject,
+  OnDestroy,
+  QueryList,
+} from '@angular/core';
+import {Subscription} from 'rxjs/Subscription';
+import {CdkTree} from './tree';
+import {NestedNode} from './tree-data';
+import {CdkNodePlaceholder} from './placeholder';
+import {CdkTreeNode} from './node';
+
+
+/**
+ * Nested node, add children to `cdkNodePlaceholder` in template.
+ */
+@Directive({
+  selector: '[cdkNestedNode]'
+})
+export class CdkNestedNode<T extends NestedNode> implements AfterContentInit, OnDestroy {
```

Comment:
Should this be `CdkNestedTreeNode`? I'm not totally clear on the different between this and `CdkTreeNode`